### ISSUE TITLE
[Android]Scene translation failed to take PixelRatio.get() into consideration

### ISSUE THREAD

**Issue by ldn0x7dc** [facebook/react-native#4221]
I'm using react-native 0.14.2.
I'm applying `Navigator.SceneConfigs.FloatFromRight` to all scenes. The bug only occurred on Android. On iOS it's OK.
Below is the related code, from file **NavigatorSceneConfigs.js**, which is shared by iOS and Android:
```javascript
var FromTheRight = {
  opacity: {
    value: 1.0,
    type: 'constant',
  },

  transformTranslate: {
    from: {x: Dimensions.get('window').width, y: 0, z: 0},
    to: {x: 0, y: 0, z: 0},
    min: 0,
    max: 1,
    type: 'linear',
    extrapolate: true,
    round: PixelRatio.get(),
  },

  translateX: {
    from: Dimensions.get('window').width,
    to: 0,
    min: 0,
    max: 1,
    type: 'linear',
    extrapolate: true,
    round: PixelRatio.get(),
  },
};
```
On an Android device, on which `PixelRatio.get()` is 3, the above translation only cover 1/3 width of the screen. Obviously this is a **BUG** and this bug did **NOT** occur with react-native 0.12.0(around, I'm not sure).
I believe this bug should be fixed in some other file, not `NavigatorSceneConfigs.js`.

**Comment by ldn0x7dc** [comment]
I tried to narrow down the file where bug is introduced.
In file 
```javascript
_transitionSceneStyle: function(fromIndex, toIndex, progress, index) {
    var viewAtIndex = this.refs['scene_' + index];
    if (viewAtIndex === null || viewAtIndex === undefined) {
      return;
    }
    // Use toIndex animation when we move forwards. Use fromIndex when we move back
    var sceneConfigIndex = fromIndex < toIndex ? toIndex : fromIndex;
    var sceneConfig = this.state.sceneConfigStack[sceneConfigIndex];
    // this happens for overswiping when there is no scene at toIndex
    if (!sceneConfig) {
      sceneConfig = this.state.sceneConfigStack[sceneConfigIndex - 1];
    }
    var styleToUse = {};
    var useFn = index < fromIndex || index < toIndex ?
      sceneConfig.animationInterpolators.out :
      sceneConfig.animationInterpolators.into;
    var directionAdjustedProgress = fromIndex < toIndex ? progress : 1 - progress;
    var didChange = useFn(styleToUse, directionAdjustedProgress);
    if (didChange) {
      viewAtIndex.setNativeProps({style: styleToUse});
    }
  },
```
Everything is OK until the last line `viewAtIndex.setNativeProps({style: styleToUse})`. I printed the data in `styleToUse`:
> {"opacity":1,"translateX":360,"scaleX":1,"scaleY":1,"transformMatrix":[1,0,0,0,0,1,0,0,0,0,1,0,360,0,0,1]}
I'm using a Samsung S6, so the screen width is 360dp, or 360x3 pixels. I think the problem is in function *setNativeProps()*, which mistake 360 as 360 pixels, rather than 360dp.

**Comment by satya164** [comment]
#3669 #4132

**Comment by 6i6arka** [comment]
@ldn0x7dc Did you manage to fix this? If yes, could you share how? I'm experiencing the same problem.

**Comment by ldn0x7dc** [comment]
@satya164 Pls check https://github.com/dgladkov/react-native/commit/590e36f478a5bd61899a28d5a0dccff6500ce4a9 
and https://github.com/facebook/react-native/pull/3954.
Seems this issue has got fixed but NOT yet available in 0.15.0. I guess the next release will fix it. 
Or if in hurry you can follow above links and modify the related java file by yourself.
